f5-openstack-agent-dist/Docker/redhat/install_test/fetch_and_install_deps.py

```diff
@@ -39,35 +45,43 @@ def fetch_agent_dependencies(dist_dir, version, release):
         print "Success"
 
     # Get the sdk requirement.
-    requiresCmd = "rpm -qRp %s/rpms/build/%s" % (dist_dir, agent_pkg)
-    print "Getting dependencies for %s." % (agent_pkg)
+    requiresCmd = "rpm -qRp %s" % agent_pkg
+    agent_pkg_base = os.path.basename(agent_pkg)
+    print "Getting dependencies for %s." % agent_pkg_base
     (output, status) = runCommand(requiresCmd)
 
     if status != 0:
-        print "Can't get package dependencies for %s" % (agent_pkg)
+        print "Can't get package dependencies for %s" % agent_pkg_base
         return 1
     else:
         print "Success"
-    
-    for line in output.split('\n'):
-        m = f5_sdk_pattern.match(line)
-        if m:
-            f5_sdk_version = m.group(1)
-            break
 
+    for line in output.split('\n'):
+        print line, dep_match_re.pattern
+        match = dep_match_re.match(line)
+        if match:
+            groups = list(match.groups())
+            my_dep = ReqDetails(groups[0], groups[2], groups[3])
```

Comment:
I stumbled upon this too.  Python2.7's re (at least) is behaving a little differently than I would expect here, and I had to work through it in the debugger.  This was what I found:

Normally, groups[0] would be the whole match; however, in this case because I'm doing a match within a match, I get:
>>> dep_match_re = re.compile('^((python|f5-sdk)[\w\-]*)\s([<>=]{1,2})\s(\S+)')
>>> dependent = 'python-six >= 1.10.2'
>>> match = dep_match_re.search(dependent)
>>> match
<_sre.SRE_Match object at 0x100e6cca8>
>>> match.groups()
('python-six', 'python', '>=', '1.10.2')
>>>
0: 'python-six' is given by the ^((python|f5-sdk)[\w\-]+)
1: 'python' is given by the internal expression (python|f5-sdk)
2: '>=' is given by the ([<>=]{1,2})
3: (\S+)

Python acts a little differently when you're doing match within a match... it's a little hairy and unfortunate that it behaves this way.  I would, like what you're implying, expect it to be 0 - the full match 1 - the python-six 2 - the python|f5-sdk matching.  It normally behaves this way, so for it to be different for a match-within-a-match was confusing.